src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -269,6 +271,19 @@ public SamReader open(final SamInputResource resource) {
                             bufferedStream.close();
                             primitiveSamReader = new CRAMFileReader(sourceFile, indexFile, referenceSource, validationStringency);
                         }
+                    } else if (SamStreams.isSRAFile(bufferedStream)) {
+                        if (sourceFile == null || !sourceFile.isFile()) {
+                            sourceFile = null;
+                        } else {
+                            bufferedStream.close();
+                            bufferedStream = null;
+                        }
+                        // Handle case in which file is a named pipe, e.g. /dev/stdin or created by mkfifo
+                        if (referenceSource != null) {
+                            primitiveSamReader = new SRAFileReader(sourceFile);
+                        } else {
+                            primitiveSamReader = new SRAFileReader(sourceFile);
+                        }
```

Comment:
This is puzzling -- your `if` and `else` clauses are identical. Collapse these two cases into one.